src/os/filestore/FileStore.cc

```diff
@@ -3329,8 +3329,13 @@ int FileStore::_write(const coll_t& cid, const ghobject_t& oid,
 
   // write
   r = bl.write_fd(**fd, offset);
-  if (r == 0)
-    r = bl.length();
+  if (r < 0) {
+    dout(0) << "write error " 
+            << cpp_strerror(r) << dendl;
+    goto close_fd;
```

Comment:
Well, don't think we need to split this statement into multiple lines since it is obviously within 80 characters, right?